x-pack/plugins/task_manager/server/task.ts

```diff
@@ -47,7 +48,7 @@ export type SuccessfulRunResult = {
    * recurring task). See the RunContext type definition for more details.
    */
   state: Record<string, unknown>;
-  hasError?: boolean;
+  taskRunError?: DecoratedError;
```

Comment:
I know this is not new but could we leave a comment about when we would see this error in the success result? Below it says the `error` in a `FailedRunResult` indicates the task failed to accomplish its work. Maybe we could leave a comment that says this `error` in a `SuccessfulRunResult` occurs when the task provides custom error handling and logging?

Reply:
WDYT of `internalError` or `internalTaskError` instead of `taskRunError`? Not sure I love either TBH